beacon-chain/blockchain/chain_info.go

```diff
@@ -277,6 +285,8 @@ func (s *Service) CurrentFork() *ethpb.Fork {
 
 // IsCanonical returns true if the input block root is part of the canonical chain.
 func (s *Service) IsCanonical(ctx context.Context, blockRoot [32]byte) (bool, error) {
+	s.ForkChoicer().RLock()
+	defer s.ForkChoicer().RUnlock()
 	// If the block has not been finalized, check fork choice store to see if the block is canonical
 	if s.cfg.ForkChoiceStore.HasNode(blockRoot) {
```

Comment:
In part I am guilty of this since I never liked the overhead of an extra call to `ForkChoicer()`, it would be nice to clean this up and actually not export the field and only use the getter.